src/lib/rnp.cpp

```diff
@@ -4391,6 +4461,54 @@ rnp_key_get_expiration(rnp_key_handle_t handle, uint32_t *result)
     return RNP_SUCCESS;
 }
 
+rnp_result_t
+rnp_key_get_revocation_reason(rnp_key_handle_t handle, char **result)
+{
+    if (!handle || !result) {
+        return RNP_ERROR_NULL_POINTER;
+    }
+    pgp_key_t *key = get_key_prefer_public(handle);
+    if (!key || !key->revoked) {
+        return RNP_ERROR_BAD_PARAMETERS;
+    }
+
+    *result = strdup(key->revocation.reason ? key->revocation.reason : "");
```

Comment:
We should probably check for `NULL` so we can return `RNP_ERROR_OUT_OF_MEMORY`.

Reply:
@dewyatt Yeah, you are right. Will fix it. Any other comments?

Reply:
It might make sense to add a note that an `rnp_uid_handle_t` can be invalidated if a userid is removed (though we may not have a function to do this *yet*). But LGTM anyways.

Reply:
@dewyatt Done, please see the latest commit.